modules/service/src/main/java/org/eclipse/xpanse/service/impl/DatabaseRegisterServiceStorage.java

```diff
@@ -0,0 +1,181 @@
+        boolean providerFilter = StringUtils.isNotBlank(query.getCspName());
+        boolean serviceNameFilter = StringUtils.isNotBlank(query.getServiceName());
+        List<RegisterServiceEntity> serviceStatusEntities = services();
+        if (!(providerFilter || serviceNameFilter)) {
+            return serviceStatusEntities;
+        }
+        List<RegisterServiceEntity> result = new ArrayList<>();
+
+        for (RegisterServiceEntity serviceStatusEntity : serviceStatusEntities) {
+            boolean isMatched = true;
+            if (providerFilter) {
+                isMatched = StringUtils.equalsIgnoreCase(query.getCspName(),
+                        serviceStatusEntity.getOcl().getCloudServiceProvider().getName().name());
+            }
+            if (serviceNameFilter) {
+                isMatched = isMatched && StringUtils.equals(query.getServiceName(),
+                        serviceStatusEntity.getOcl().getName());
+            }
+            if (isMatched) {
+                result.add(serviceStatusEntity);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Method to get database entry based registerServiceEntity.
+     *
+     * @param uuid uuid of registerServiceEntity.
+     * @return Returns the database entry for the provided arguments.
+     */
+    @Override
+    public RegisterServiceEntity getRegisterServiceById(UUID uuid) {
+        Optional<RegisterServiceEntity> option = registerServiceRepository.findById(uuid);
+        if (option.isPresent()) {
+            return option.get();
+        } else {
+            throw new IllegalStateException(
+                    String.format("Registered service %s not existed.", uuid));
+        }
+    }
+
+    /**
+     * Method to get all stored database entries.
+     *
+     * @return Returns all rows from the service status database table.
+     */
+    @Override
+    public List<RegisterServiceEntity> services() {
+        return registerServiceRepository.findAll();
+    }
+
+    /**
+     * Remove register service entity from database by uuid.
+     *
+     * @param uuid uuid of register service entity
+     */
+    @Override
+    public void removeById(UUID uuid) {
+        if (registerServiceRepository.existsById(uuid)) {
+            registerServiceRepository.deleteById(uuid);
+        } else {
+            throw new IllegalStateException(
+                    String.format("Registered service %s not existed.", uuid));
+        }
+
+    }
+
+    /**
+     * Remove register service entity from database by entity.
+     *
+     * @param registerServiceEntity register service entity
+     */
+    @Override
+    public void remove(RegisterServiceEntity registerServiceEntity) {
+        registerServiceRepository.delete(registerServiceEntity);
+    }
+
+
+    private String getRegisterServicePrimaryKey(RegisterServiceEntity registerService) {
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Primary key is the UUID in the entity. Do we change it to a different one now?

Reply:
This method is used to find the registered service with the same service name, version, service version and service provider in the database when a new service is registering.

Reply:
No. This method is just used to find the registered service with the same service name, version, service version and service provider in the database when a new service is registering.  If the same registered service is found, throw exception to show the service already registered.

Reply:
ok understood. then we can rename the method to something else which does not involve 'primary key'.

Reply:
Yes. I will change the method name.